device_doctor/lib/src/android_device.dart

```diff
@@ -45,26 +43,8 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
 
   Future<List<String>> _deviceListOutputWithRetries(Duration retryDuration, {ProcessManager processManager}) async {
     const Duration deviceOutputTimeout = Duration(seconds: 15);
-    RetryOptions r = RetryOptions(
-      maxAttempts: 3,
-      delayFactor: retryDuration,
-    );
-    return await r.retry(
-      () async {
-        String result = await _deviceListOutput(deviceOutputTimeout, processManager: processManager);
-        return result.trim().split('\n');
-      },
-      retryIf: (Exception e) => e is TimeoutException,
-      onRetry: (Exception e) => _killAdbServer(processManager: processManager),
-    );
-  }
-
-  void _killAdbServer({ProcessManager processManager}) async {
-    if (Platform.isWindows) {
-      await killAllRunningProcessesOnWindows('adb', processManager: processManager);
-    } else {
-      await eval('adb', <String>['kill-server'], canFail: false, processManager: processManager);
-    }
+    String result = await _deviceListOutput(deviceOutputTimeout, processManager: processManager);
```

Comment:
What are we trying to accomplish with this change? removing the retries will probably make things more flaky

Reply:
Can we keep the retry but remove the kill server?

Can we somehow detect if a test is running on the device during this?

Can we move this code from infra to a setup/teardown step on the test(s)?

Reply:
I believe this runs only when the bot is idle, are we running it somewhere else?

Reply:
Intended to remove the `adb kill-server` to avoid interrupting test execution.

However, after revisiting the chromium log: `get_dimensions` or `get_state` only runs when no test is running. So this should not be the culprit of issue: https://github.com/flutter/flutter/issues/93661.

Closing.

Reply:
> I believe this runs only when the bot is idle, are we running it somewhere else?

(didn't see this comment before posting)

Currently we have only two parts running `adb kill-server`: one in `get_dimension`, the other in `get_state`. Doesn't sound caused by device doctor. Wondering if anything else may cause `adb logcat` dying.

Reply:
If this is happening only on adb logcat then my guess is that the device is crashing. Is there a particular test where this is happening?